src/Microsoft.DotNet.Build.Tasks/PackageFiles/init-tools.ps1

```diff
@@ -19,3 +19,47 @@ foreach ($file in Get-ChildItem $ToolRuntimePath *.runtimeconfig.json)
 # the init-tools.cmd (that is checked into each repository that uses buildtools) can write the semaphore
 # marker into this file once tool initialization is complete.
 New-Item -Force -Type Directory (Join-Path $ToolRuntimePath (Split-Path -Leaf (Split-Path $BuildToolsPackageDir)))
+
+# Download the package version props file, if  was passed in the environment.
+$packageVersionPropsUrl = $env:PACKAGEVERSIONPROPSURL
+$packageVersionPropsPath = Join-Path $ToolRuntimePath "DownloadedPackageVersions.props"
+
+if ($packageVersionPropsUrl)
+{
+    Write-Host "Downloading package version props from '$packageVersionPropsUrl' to '$packageVersionPropsPath'..."
+
+    # Copied from init-tools.cmd in CoreFX
+    $retryCount = 0
+    $success = $false
+    $proxyCredentialsRequired = $false
+    do
+    {
+        try
+        {
+            $wc = New-Object Net.WebClient
```

Comment:
While I don't like the duplicated logic for downloading things both here and in the init-tools scripts in the repo's I guess we can live with it until we get our auto-updating scripts workflow going and this can be shared.

Reply:
Yeah, pained me to put this in here but I think it's the reasonable way in the short term while keeping the retries for reliability.